src/util/__tests__/importRoam.ts

```diff
@@ -0,0 +1,271 @@
+      - test_create@yahoo.com
+    - =edit-email
+      - test_edit@yahoo.com
+  - Banana
+    - =create-email
+      - test_create@icloud.com
+    - =edit-email
+      - test_edit@icloud.com
+- Veggies
+  - Broccoli
+    - =create-email
+      - test_create@gmail.com
+    - =edit-email
+      - test_edit@gmail.com
+  - Spinach
+    - =create-email
+      - test_create@icloud.com
+    - =edit-email
+      - test_edit@icloud.com
+`)
+})
+
+test('it should convert a Roam json into a list of thoughts and subthoughts with correct indentation', () => {
+  const testData = [
+    {
+      title: 'September 4th, 2020',
+      children: [
+        {
+          string: 'A',
+          'create-email': 'test_create@gmail.com',
+          'create-time': 1600111381583,
+          children: [
+            {
+              string: 'B',
+              'create-email': 'test_create@gmail.com',
+              'create-time': 1600111383054,
+              children: [
+                {
+                  string: 'C',
+                  'create-email': 'test_create@gmail.com',
+                  'create-time': 1600111383911,
+                  uid: 'HMN_YQtZZ',
+                }],
+              uid: 'JBXKlMcxh',
+            }],
+          uid: '0VQBPmUiy',
+        }],
+    },
+    {
+      title: 'September 5th, 2020',
+      children: [
+        {
+          string: 'X',
+          'create-email': 'test_create@gmail.com',
+          'create-time': 1600111456859,
+          children: [
+            {
+              string: 'Y',
+              'create-email': 'test_create@gmail.com',
+              'create-time': 1600111457621,
+              children: [
+                {
+                  string: '[[September 4th, 2020]]',
+                  'create-email': 'test_create@gmail.com',
+                  'create-time': 1600111458385,
+                  uid: 'Wt5NR3b56',
+                }],
+              uid: 'obXRTMWqJ',
+            }],
+          uid: 'Pu444IoIi',
+        }],
+    }]
+  const res = importExport(testData)
+  expect(res)
+    .toBe(`
+- September 4th, 2020
+  - A
+    - B
+      - C
+        - =create-email
+          - test_create@gmail.com
+      - =create-email
+        - test_create@gmail.com
+    - =create-email
+      - test_create@gmail.com
+- September 5th, 2020
+  - X
+    - Y
+      - [[September 4th, 2020]]
+        - =create-email
+          - test_create@gmail.com
+      - =create-email
+        - test_create@gmail.com
+    - =create-email
+      - test_create@gmail.com
+`)
+})
+
+test('it should save create-time as created property', () => {
+  const roamBlocks = testData.map(roamBlock => roamBlock.children).flat()
+
+  const {
+    contextIndexUpdates: contextIndex,
+    thoughtIndexUpdates: thoughtIndex,
+  } = importRoam(testState, RANKED_ROOT as SimplePath, testData)
+
+  Object.keys(contextIndex)
+    .forEach(contextHash => {
+      expect(contextIndex[contextHash].lastUpdated).toEqual('2020-11-02T01:11:58.869Z')
+    })
```

Comment:
@anmolarora1 I found an even better way to do this. Using `toMatchObject` we can assert arbitrary keys in a single assertion. This obviates the need for asserting the number of keys and has the advantage of displaying all values that fail instead of just the first one that fails in the `forEach` loop. See: https://github.com/cybersemics/em/blob/staged-loading/src/util/__tests__/roamJsonToBlocks.ts#L226-L237

The object can be built up abstractly (i.e. `reduce`), but here I prefer the hard-coded values as they are easier to read and make it easier to assert exceptions, such as `Fruits` and `Veggies` using the edit time of their child.